devicemodel/hw/pci/ivshmem.c

```diff
@@ -272,9 +272,10 @@ pci_ivshmem_init(struct vmctx *ctx, struct pci_vdev *dev, char *opts)
 		pr_warn("the shared memory size is incorrect, %s\n", tmp);
 		goto err;
 	}
+	size *= 0x100000; /* convert to megabytes */
 	if (size < 0x200000 || size >= 0x40000000 ||
 			(size & (size - 1)) != 0) {
-		pr_warn("invalid shared memory size %u, the size range is [2M,1G) bytes and value must be a power of 2\n",
+		pr_warn("Invalid shared memory size %u, the size range is [2MB,512MB], the unit is megabyte and the value must be a power of 2\n",
```

Comment:
The value of ``size`` was converted to megabytes on line 275, so the value shown in this error message is no longer the value specified by the user.  Change the value in this error message to be divided by 0x100000.